src/util.ts

```diff
@@ -89,6 +89,30 @@ export function createOutputRetainingCompilerHost(
   }
 }
 
+/**
+ * Returns the declarations for the given decorator.
+ */
+export function getDecoratorDeclarations(decorator: ts.Decorator, typeChecker: ts.TypeChecker) {
+  // Walk down the expression to find the identifier of the decorator function.
+  let node: ts.Node = decorator;
+  while (node.kind !== ts.SyntaxKind.Identifier) {
+    switch (node.kind) {
+      case ts.SyntaxKind.Decorator:
+      case ts.SyntaxKind.CallExpression:
+        node = (node as ts.Decorator | ts.CallExpression).expression;
+        break;
+      default:
+        throw new Error();
```

Comment:
Your comment makes a lot of sense. I made sure not to throw on decorator types that the code can't (yet) handle.